src/JBDelegateMetadataLib.sol

```diff
@@ -74,4 +74,91 @@ library JBDelegateMetadataLib {
         }
     }
 
+
+
+    /**
+     * @notice Add a delegate to an existing metadata
+     *
+     * @param _idToAdd         The id of the delegate to add
+     * @param _dataToAdd       The metadata of the delegate to add
+     * @param _originalMetadata The original metadata
+     *
+     * @return _newMetadata    The new metadata with the delegate added
+     */
+    function addToMetadata(bytes4 _idToAdd, bytes calldata _dataToAdd, bytes calldata _originalMetadata) public pure returns (bytes memory _newMetadata) {
+        // Get the first data offset - upcast to avoid overflow (same for other offset)...
+        uint256 _firstOffset = uint8(_originalMetadata[RESERVED_SIZE + ID_SIZE]);
```

Comment:
wondering since both the lib and contract share same constants we can put that in a common file and import